### PR TITLE
Add new module gatk4/gatherbqsr

### LINKED ISSUE #199: new module: gatk4/gatherbqsrreports
I think it would be good to have module for [gatk4](https://bioconda.github.io/recipes/gatk4/README.html)

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

Closes #199  

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/gatk4/gatherbqsrreports/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.2.3.0--hdfd78af_1':
+        'quay.io/biocontainers/gatk4:4.2.3.0--hdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(recal_table)
+
+    output:
+    tuple val(meta), path("*.table"), emit: table
+    path "versions.yml"                   , emit: versions
```
Comment (maxulysse) [inline modules/gatk4/gatherbqsrreports/main.nf:15]:
```suggestion
    path "versions.yml"             , emit: versions
```

**Comment by maxulysse** [comment]
What happened to `tests/config/pytest_modules.yml`?

File: tests/modules/gatk4/gatherbqsrreports/main.nf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+workflow test_gatk4_gatherbqsrreports_multiple {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        [file(params.test_data['homo_sapiens']['illumina']['test_baserecalibrator_table'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test2_baserecalibrator_table'], checkIfExists: true)
+        ]
+]
```
Comment (mahesh-panchal) [inline tests/modules/gatk4/gatherbqsrreports/main.nf:24]:
```suggestion
    input = [
        [ id:'test', single_end:false ], // meta map
        [ 
            file(params.test_data['homo_sapiens']['illumina']['test_baserecalibrator_table'], checkIfExists: true),
            file(params.test_data['homo_sapiens']['illumina']['test2_baserecalibrator_table'], checkIfExists: true)
        ]
    ]
```

File: modules/gatk4/gatherbqsrreports/main.nf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    gatk --java-options "-Xmx${avail_mem}g" \\
+        GatherBQSRReports \
+        ${input} \
+        --tmp-dir . \
+        $args \
+        --output ${prefix}.table
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        gatk4: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$//')
```
Comment (mahesh-panchal) [inline modules/gatk4/gatherbqsrreports/main.nf:38]:
```suggestion
        gatk4: \$( gatk --version |& sed 's/^.*(GATK) v//; s/ .*\$//' )
```
This is a minor thing, but for me the `echo` is unnecessary.

Comment (FriederikeHanssen) [reply modules/gatk4/gatherbqsrreports/main.nf:38]: ok, I c&p it from other gatk modules. we can remove it there then as well.

**Comment by FriederikeHanssen** [comment]
> What happened to `tests/config/pytest_modules.yml`?

No idea 😱 I'll take a look.

**Comment by FriederikeHanssen** [comment]
ok can't figure it out. I'll be lazy close this PR and open a new one

**Pull request closed without merging**